GoogleMapsComponents/Maps/BaseListableEntity.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.JSInterop;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GoogleMapsComponents.Maps
+{
+    public abstract class BaseListableEntity<T> : IDisposable, IJsObjectRef
+        where T: BaseListableEntityOptions
+    {
+        protected readonly JsObjectRef _jsObjectRef;
+
+        public readonly Dictionary<string, List<MapEventListener>> EventListeners;
+
+        public Guid Guid => _jsObjectRef.Guid;
+
+        internal BaseListableEntity(JsObjectRef jsObjectRef)
```

Comment:
I am not sure but i think there was issues with internal constructor in some class. Will change if needed.

Reply:
Really?
Can you explain in detail?

Reply:
In any case, let me introduce myself: I am almost 50 years old and since I was 15 I started developing software (it was the time of the IMB M20 with green phosphor monitor). I have worked with basic, gwbasic, Delphi, PLC, VxWorks, Sun Os, HPUx, ASP, VB6, Com Interopt, MSMQ, PHP, .Net, AS400, Oracle, SQL Server, MySQL, Interbase, DB2, bash, NodeJs, IoT, etc.
I am Oracle DBA and SQL server and Team Account Manager, but by virtue of the many things I have worked on, I am increasingly being used as a "last resort" to fix "disasters", so my working life is a continuous "problem solving" , restoration of corrupt databases, revision of wrong SW architectures (often not only in terms of implementation, but precisely in the choice of paradigm).
This is why I have a very active mind (and I consider myself lucky at this age), but for the same reason I am used to the quick solution, and in spite of myself I no longer focus (for lack of time) on the "aesthetic beauty" of the code and you don't know how sad it is for me because programming has always been an art form for me.
So I hope you want to apologize for the many distractions committed :)
And lastly, after decades of writing legacy code, I became passionate about open source projects and when I find one like this, if I have the strength and the time, I like to participate.
From now on I will try to use my free time with the approach of passion and not that of the "problem solver": the quality of the code will certainly be better! :)

Reply:
Sorry it was vice versa. You made change previous;y "Due to need of implementation, I must define previous private constructor for Marker as", so my bad.
I remember once read comment on Hacker News, "How you could do programming on your free time? It is nonsense". I think real passionate programmer love to spend time on free time. It is almost like a hobby. I am you side as i also like to spend time in front of screen when i have free time.

Speaking about "aesthetic beauty" i think it is more important to make it nice in new solutions, but yea on legacy it overkill of time spending. Aslo once read article about screen reader and programming for these guys it is very helpful. Maybe not related "26 percent (one in 4) of adults in the United States have some type of disability".
And finally as always. There is no golden bullet. We share, discuss together to archive best. Thank you for you time.